utils.js

```diff
@@ -585,21 +585,38 @@ function makeDefaults(html, userID) {
   };
 }
 
-function parseAndCheckLogin(jar, defaultFuncs) {
+function parseAndCheckLogin(jar, defaultFuncs, retryCount) {
+  if (retryCount == undefined) {
+    retryCount = 0;
+  }
   return function(data) {
     return bluebird.try(function() {
       log.verbose("parseAndCheckLogin", data.body);
       if (data.statusCode >= 500 && data.statusCode < 600) {
-        log.warn("parseAndCheckLogin", "Got status code " + data.statusCode + " retrying...");
+        if (retryCount == 5) {
+          throw {
+            error: "Request retry failed. Check the `res` and `statusCode` property on this error.",
+            statusCode: data.statusCode,
+            res: data.body
+          };
+        }
+        retryCount++;
+        log.warn("parseAndCheckLogin", "Got status code " + data.statusCode + " - " + retryCount + ". attempt to retry in 5 seconds...");
         var url = data.request.uri.protocol + "//" + data.request.uri.hostname + data.request.uri.pathname;
         if (data.request.headers['Content-Type'].split(";")[0] === "multipart/form-data") {
-          return defaultFuncs
-            .postFormData(url, jar, data.request.formData, {})
-            .then(parseAndCheckLogin(jar, defaultFuncs));
+          return bluebird
+            .delay(5000)
+            .then(function() {
+              return defaultFuncs.postFormData(url, jar, data.request.formData, {});
+            })
+            .then(parseAndCheckLogin(jar, defaultFuncs, retryCount));
         } else {
-          return defaultFuncs
-            .post(url, jar, data.request.formData)
-            .then(parseAndCheckLogin(jar, defaultFuncs));
+          return bluebird
+            .delay(5000)
+            .then(function() {
+              return post(url, jar, data.request.formData);
```

Comment:
Yeah, I think we want to be using this https://github.com/Schmavery/facebook-chat-api/blob/830db449716f49481da8e9b34544b37abe7e0cf3/utils.js#L583 (defaultFuncs.post).  Sorry, this is a little confusing because of the utils.post function (which doesn't add some default values)